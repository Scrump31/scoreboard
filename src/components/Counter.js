import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

const Counter = ({ id, score, updateScore }) => {
  const addScore = () => updateScore(score + 1, id);
  const subtractScore = () => updateScore(score - 1, id);

  return (
    <div>
      <Button bsStyle="danger" bsSize="large" onClick={subtractScore}>
        -
      </Button>
      <span className="player-score">{score}</span>
      <Button bsStyle="success" bsSize="large" onClick={addScore}>
        +
      </Button>
    </div>
  );
};

Counter.propTypes = {
  id: PropTypes.number,
  score: PropTypes.number,
  updateScore: PropTypes.func.isRequired,
}.isRequired;

export default Counter;
