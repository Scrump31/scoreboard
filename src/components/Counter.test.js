import React from 'react';
import { shallow, mount } from 'enzyme';
import Counter from './Counter';

const ADD_BTN = 'button.btn-success';
const SUBTRACT_BTN = 'button.btn-danger';
const SCORE = 'span';

describe('<Counter />', () => {
  test('renders without crashing', () => {
    const component = shallow(<Counter />);
    expect(component).toMatchSnapshot();
  });

  it('displays a score based on props.score', () => {
    const component = shallow(<Counter score={3} />);
    expect(component.find(SCORE).text()).toEqual('3');
  });

  it('calls mockUpdateScore() when "Add" button clicked', () => {
    const mockUpdateScore = jest.fn();
    const wrapper = mount(<Counter updateScore={mockUpdateScore} />);
    wrapper.find(ADD_BTN).simulate('click');
    expect(mockUpdateScore).toHaveBeenCalled();
    wrapper.unmount();
  });

  it('calls mockUpdateScore() when "Subtract" button clicked', () => {
    const mockUpdateScore = jest.fn();
    const wrapper = mount(<Counter updateScore={mockUpdateScore} />);
    wrapper.find(SUBTRACT_BTN).simulate('click');
    expect(mockUpdateScore).toHaveBeenCalled();
    wrapper.unmount();
  });
});
