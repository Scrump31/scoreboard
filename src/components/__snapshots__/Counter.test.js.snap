// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders without crashing 1`] = `
<div>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="large"
    bsStyle="danger"
    disabled={false}
    onClick={[Function]}
  >
    -
  </Button>
  <span
    className="player-score"
  />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="large"
    bsStyle="success"
    disabled={false}
    onClick={[Function]}
  >
    +
  </Button>
</div>
`;
